src/codegen/dispatch/solana.rs

```diff
@@ -445,6 +444,57 @@ fn add_constructor_dispatch_case(
     let entry = cfg.new_basic_block(format!("constructor_cfg_{cfg_no}"));
     cfg.set_basic_block(entry);
 
+    // check for magic in data account, to see if data account is initialized
+    let magic_ok = cfg.new_basic_block("magic_ok".into());
+    let magic_bad = cfg.new_basic_block("magic_bad".into());
+
+    cfg.add(
+        vartab,
+        Instr::LoadStorage {
+            res: magic,
+            ty: Type::Uint(32),
+            storage: Expression::NumberLiteral {
+                loc: Loc::Codegen,
+                ty: Type::Uint(32),
+                value: 0.into(),
+            },
+        },
+    );
+
+    cfg.add(
+        vartab,
+        Instr::BranchCond {
+            cond: Expression::Equal {
+                loc: Loc::Codegen,
+                left: Expression::Variable {
+                    loc: Loc::Codegen,
+                    ty: Type::Uint(32),
+                    var_no: magic,
+                }
+                .into(),
+                right: Expression::NumberLiteral {
+                    loc: Loc::Codegen,
+                    ty: Type::Uint(32),
+                    value: 0.into(),
+                }
+                .into(),
+            },
+            true_block: magic_ok,
+            false_block: magic_bad,
+        },
+    );
+
+    cfg.set_basic_block(magic_bad);
+
+    cfg.add(
+        vartab,
+        Instr::ReturnCode {
+            code: ReturnCode::InvalidDataError,
+        },
+    );
+
+    cfg.set_basic_block(magic_ok);
```

Comment:
You have here 48 lines of repeated code. What do you think of creating a function?

Reply:
Good point, fixed